kinto_ldap/authentication.py

```diff
@@ -20,18 +20,44 @@ def user_checker(username, password, request):
     settings = request.registry.settings
     cache_ttl = settings['ldap.cache_ttl_seconds']
     hmac_secret = settings['userid_hmac_secret']
-    cache_key = utils.hmac_digest(hmac_secret, '%s:%s' % (username, password))
+    cache_key = utils.hmac_digest(hmac_secret, '{}:{}'.format(username, password))
 
     cache = request.registry.cache
     cache_result = cache.get(cache_key)
 
-    ldap_fqn = settings['ldap.fqn']
+    bind_dn = settings.get('ldap.bind_dn')
+    bind_password = settings.get('ldap.bind_password')
 
     if cache_result is None:
         cm = request.registry.ldap_cm
 
+        # 0. Generate a search filter by combining the attribute and
+        # filter provided in the ldap.fqn_filters directive with the
+        # username passed by the HTTP client.
+        base_dn = settings['ldap.base_dn']
+        filters = settings['ldap.filters'].format(mail=username)
```

Comment:
note: this will fail if `{mail}` is not in the filters setting. Should we raise a `ConfigurationError` in case if it's not?